torch/_subclasses/fake_tensor.py

```diff
@@ -1538,6 +1538,19 @@ def _validate_cache_key(
             raise _BypassDispatchCache("data dependent output")
 
         if torch.Tag.dynamic_output_shape in func.tags:
+            if func is aten.index.Tensor:
```

Comment:
Is the reason that index.Tensor does not have dynamic output shape in this situation? Can you write that in a comment and/or the PR body if so?